src/main/java/ConnectionManagement/ConnectionManager.java

```diff
@@ -0,0 +1,161 @@
+package ConnectionManagement;
+
+import java.io.Closeable;
+import java.util.ArrayList;
+
+/**
+ * Created by alfatihmukhtar on 2/3/17.
+ */
+public class ConnectionManager {
+    // FIELDS
+    private int maxConnections;
+    private ArrayList<ManagedConnection> activeConnections = new ArrayList<ManagedConnection>();
+
+    // CONSTRUCTORS
+    public ConnectionManager() { maxConnections = 5; }
+    public ConnectionManager(int limit) {
+        maxConnections = limit;
+    }
+
+    // GETTERS
+    public String getActiveConnectionsLine() {
+        String totalConnections = "";
+        for(ManagedConnection connection : activeConnections) {
+            totalConnections+=(connection.connect()+"\n");
+        }
+        return totalConnections;
+    }
+    public ArrayList<ManagedConnection> getActiveConnections() {
+        return activeConnections;
+    }
+
+    // METHODS: Making Connection objects.
+    public void makeConnection() {
+        if(hasRoomForConnections()) {
+            ManagedConnection connection = this.new ManagedConnection();
+            activeConnections.add(connection);
+        } else {
+            reachedMaxConnections();
+        }
+    }
+    public void makeConnection(String protocol) {
+        if(hasRoomForConnections()) {
+            ManagedConnection connection = this.new ManagedConnection(protocol);
+            activeConnections.add(connection);
+        } else {
+            reachedMaxConnections();
+        }
+    }
+    public void makeConnection(Integer port) {
+        if(hasRoomForConnections()) {
+            ManagedConnection connection = new ManagedConnection(port);
+            activeConnections.add(connection);
+        } else {
+            reachedMaxConnections();
+        }
+    }
+
+    // METHODS: Close a connection by index.
+    public void closeConnection(int index) {
+        activeConnections.get(index).close();
+        activeConnections.remove(index);
+    }
+
+    // METHODS: Helper methods.
+    public boolean hasRoomForConnections() {
+        return activeConnections.size()<maxConnections;
+    }
+    private void reachedMaxConnections() {
+        System.out.println("Sorry, but you cannot add more connections to this connection manager. " +
+                "You have reached the limit.");
+    }
+
+    // INNER CLASS
+    public class ManagedConnection implements Connection, Closeable {
```

Comment:
According to the spec, ManagedConnection should be private. Any classes outside of ConnectionManager that use it should only be aware of the fact that they're working with Connection or Closeable objects.